### ISSUE TITLE
Error sending response to the private room -- unverified device

### ISSUE THREAD

**Issue by saket424** [Skylar-Tech/node-red-contrib-matrix-chat#118]
@skylord123 
I created a new bot user and plugged it into the flow that auto accepts and joins the room that it was invited into

when it is supposed to respond to "users", while it successfully receives the message and decrypts it, when it tries to respond, I get this error about the existence of an unverified device (which is the bot user using nodered)

msg : string[164]
"Error sending message: UnknownDeviceError: This room contains unknown devices which have not been verified. We strongly recommend you verify them before continuing.

I am using synapse 1.113

do you recommend any remedy for this end2end encrypted private room ?

**Comment by skylord123** [comment]
#87 #93 are both related to this. There currently is no way to verify devices within this module. The matrix-js-sdk docs are quite sparse on any specifics about how to handle this. I believe I have a working version on a branch but I am not sure if the implementation is correct. 

If you want to find out more information we had a conversation about this on the matrix room here:
https://matrix.to/#/!ppQOFZPDOZdYkxhyRo:skylar.tech/$LpIQ9XmTe0VVj9TThKwLHqwfKbrAwfSdamXLTQOdkhk?via=skylar.tech&via=matrix.org&via=chat.rios.org.br

If anyone wants to help add this functionality pull requests are welcome.

**Comment by saket424** [comment]
@skylord123 
There is a new bot implementation that uses the rust sdk 
https://github.com/etkecc/baibot/blob/main/etc/app/config.yml.dist that appears to work well.
Could you check your implementation against that ?

**Comment by saket424** [comment]
There is an upcoming matrix conference that I am looking forward to attending online
https://cfp.matrix.org/matrixconf2024/talk/9BCM3A/

I believe they recommend the use of the rust crypto sdk even for non-rust uses

**Comment by saket424** [comment]
Here is a related YouTube video that is still over my head that may help others https://youtu.be/F3_Y02A53Zc?feature=shared

**Comment by skylord123** [comment]
> @skylord123 I assume you are aware of this projects that abstracts all the crypto layer to the rust sdk https://github.com/matrix-org/matrix-rust-sdk-crypto-nodejs

Yeah this is planned for next release. Created #122 for tracking this. libolm is being deprecated so we have to switch to rust crypto.

**Comment by sgofferj** [comment]
I'm using the token of a verified session from my phone and I still have the same problem. The problem appeared when I updated to the latest version and it broke basically my complete home assistant notification system.
As an intermediate solution, I would like to suggest a configuration option "Ignore unverified devices" or something like that to allow sending to a room regardless if there's unverified devices in it.

**Comment by skylord123** [comment]
@saket424 @sgofferj — v1.0.0 finally addresses this. Both pieces are now in place:

1. The Rust crypto migration you pointed at (#122) — `matrix-js-sdk` is on 41.5.0 and E2EE runs on the Rust crypto stack.
2. Device verification, built on top of it, so the `UnknownDeviceError` can be resolved by actually verifying the bot's session and the other devices, not just disabling the check.

Two ways to verify, depending on whether you want a flow or not:

- **From the server config node** — the new **Pending verification requests** button lists incoming requests and walks you through the SAS (emoji) check interactively, no flow required. The same dialog also has **Set up secure backup & cross-signing** (run this once so the bot's own session shows as verified to other clients) and **Manage sessions**.
- **From a flow** — the new `matrix-verification` (event source) and `matrix-verification-action` (request / accept / start SAS / confirm / cancel) nodes.

A v1.0.0 beta is out — please give it a try:

```
npm install node-red-contrib-matrix-chat@beta
```

Full changelog and tracking PR: #142 · release notes: https://github.com/Skylar-Tech/node-red-contrib-matrix-chat/releases/tag/v1.0.0-beta.1
